tests/test_search.py

```diff
@@ -489,3 +492,35 @@ def test_search_settings_from_root(capsys, static_mixer_settings_session):
         "conduction", search_root=solver.setup.boundary_conditions.wall["wall"].phase
     )
     assert '<search_root>["<name>"].shell_conduction["<name>"] (Object)' in results
+
+
+def test_search_whole_word_parent_child(monkeypatch, capsys):
+    api_tree_data = {
+        "api_objects": [
+            "<solver_session>.parent (Object)",
+            "<solver_session>.parent.child (Parameter)",
+        ],
+        "api_tui_objects": [],
+        "all_api_object_name_synsets": {  # why doesn't _search_whole_word() work without synsets?
```

Comment:
`why doesn't _search_whole_word() work without synsets?` - It does. We were not using synsets for `_search_whole_word()`. We were using `api_tree_data["all_api_object_synsets"].keys()` as it was a list of all API object names. 

Now we have added `all_api_object_names` separately therefore this is not relevant now.